scripts/drone/pipelines/swagger_gen.star

```diff
@@ -0,0 +1,95 @@
+"""
+This module returns all pipelines used in OpenAPI specification generation of Grafana HTTP APIs
+"""
+
+load(
+    "scripts/drone/steps/lib.star",
+    "clone_enterprise_step_pr",
+)
+load(
+    "scripts/drone/utils/images.star",
+    "images",
+)
+load(
+    "scripts/drone/utils/utils.star",
+    "pipeline",
+)
+load(
+    "scripts/drone/vault.star",
+    "from_secret",
+)
+
+def clone_pr_branch(ver_mode):
+    """Clones the PR branch, in contrast to the default cloning which checks out the target branch main and merges the changes on top.
+
+    Args:
+        ver_mode: ?
+
+    Returns:
+      Drone step.
+    """
+    if ver_mode != "pr":
+        return None
+
+    committish = "${DRONE_SOURCE_BRANCH}"
+    return {
+        "name": "clone-pr-branch",
+        "image": images["go"],
+        "commands": [
+            "apk add --update git",
+            "echo 'listing before cloning'",
+            "ls -l",
+            "cat Makefile",
+            "echo 'cloning repo'",
+            "git clone https://github.com/grafana/grafana.git",
+            "echo 'cloned repo'",
+            "cd grafana",
+            "echo 'listing...'",
+            "ls -l",
+            "pwd",
+            "cat Makefile",
+            "git checkout {}".format(committish),
+        ],
+    }
+
+def swagger_gen_step(ver_mode):
+    if ver_mode != "pr":
+        return None
+
+    committish = "${DRONE_SOURCE_BRANCH}"
+    return {
+        "name": "swagger-gen",
+        "image": images["go"],
+        "environment": {
+            "GITHUB_TOKEN": from_secret("github_token"),
+        },
+        "commands": [
+            "apk add --update git make",
+            "pwd",
+            "echo 'listing...'",
+            "ls -l",
+            "pwd",
+            "cat Makefile",
+            "make swagger-clean && make openapi3-gen",
+            "for f in public/api-spec.json public/api-merged.json public/openapi3.json; do git add $f; done",
```

Comment:
starting from #75133 the `api-spec.json` is no committed because it in [`.gitignore`](https://github.com/grafana/grafana/blob/fbbf9b1a8fa38281a0ac95e70c10b3289ca365a8/.gitignore#L42) so you can remove it 
```suggestion
            "for f in public/api-merged.json public/openapi3.json; do git add $f; done",
```